integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanChargeSpecificDueDateTest.java

```diff
@@ -233,6 +236,109 @@ public void testApplyAndWaiveLoanSpecificDueDatePenaltyWithDisbursementDate() {
+
+        // Repay the first charge fully, 10
+        Float amount = Float.valueOf("10.00");
+        transactionDate = transactionDate.plusDays(1);
+        operationDate = Utils.dateFormatter.format(transactionDate);
+        log.info("Operation date {}", transactionDate);
+        PostLoansLoanIdTransactionsResponse loanIdTransactionsResponse = loanTransactionHelper.makeLoanRepayment(operationDate, amount,
+                loanId);
+        assertNotNull(loanIdTransactionsResponse);
+        log.info("Loan Transaction Id: {} {}", loanId, loanIdTransactionsResponse.getResourceId());
+
+        // Second Loan Charge
+        feeAmount = "15.00";
+        transactionDate = transactionDate.plusDays(1);
+        operationDate = Utils.dateFormatter.format(transactionDate);
+        log.info("Operation date {}", transactionDate);
+        payloadJSON = LoanTransactionHelper.getSpecifiedDueDateChargesForLoanAsJSON(chargeId.toString(), operationDate, feeAmount);
+        postLoansLoanIdChargesResponse = loanTransactionHelper.addChargeForLoan(loanId, payloadJSON, responseSpec);
+        assertNotNull(postLoansLoanIdChargesResponse);
+        final Long loanChargeId02 = postLoansLoanIdChargesResponse.getResourceId();
+        assertNotNull(loanChargeId02);
+
+        getLoansLoanIdResponse = loanTransactionHelper.getLoan(requestSpec, responseSpec, loanId);
+        validateLoanAccount(getLoansLoanIdResponse, Double.valueOf(principalAmount), Double.valueOf("15.00"), true);
+
+        // Run Accruals
+        log.info("Running Periodic Accrual for date {}", transactionDate);
+        periodicAccrualAccountingHelper.runPeriodicAccrualAccounting(operationDate);
+        getLoansLoanIdResponse = loanTransactionHelper.getLoan(requestSpec, responseSpec, loanId);
+        loanTransactionHelper.evaluateLoanTransactionData(getLoansLoanIdResponse, "loanTransactionType.accrual", Double.valueOf("25.00"));
+
+        // Third Loan Charge
+        feeAmount = "25.00";
+        transactionDate = transactionDate.plusDays(1);
+        operationDate = Utils.dateFormatter.format(transactionDate);
+        log.info("Operation date {}", transactionDate);
+        payloadJSON = LoanTransactionHelper.getSpecifiedDueDateChargesForLoanAsJSON(chargeId.toString(), operationDate, feeAmount);
+        postLoansLoanIdChargesResponse = loanTransactionHelper.addChargeForLoan(loanId, payloadJSON, responseSpec);
+        assertNotNull(postLoansLoanIdChargesResponse);
+        final Long loanChargeId03 = postLoansLoanIdChargesResponse.getResourceId();
+        assertNotNull(loanChargeId03);
+
+        getLoansLoanIdResponse = loanTransactionHelper.getLoan(requestSpec, responseSpec, loanId);
+        validateLoanAccount(getLoansLoanIdResponse, Double.valueOf(principalAmount), Double.valueOf("40.00"), true);
+        loanTransactionHelper.evaluateLoanTransactionData(getLoansLoanIdResponse, "loanTransactionType.accrual", Double.valueOf("50.00"));
```

Comment:
Why is it 50? There was no periodic accrual execution yet...

Reply:
Would you also please check whether the very last accrual transaction amount, penalty portion, transaction date is correct and it created the journal entries for it as well?

Reply:
Fixed, there were some issues in this point, as we saw the IT was not using Accrual accounting and in the code the Fee Details was adding the PENALTIES and the read for the Accrual Transaction was reading PENALTY. That includes the old code too.